packages/@ember/runloop/index.ts

```diff
@@ -213,7 +268,31 @@ export function join() {
   @since 1.4.0
   @public
 */
-export const bind = (...curried) => {
+export function bind<
```

Comment:
One downside to trying to *actually* type `bind`, and why I ultimately chose not to on DT, is that it makes our `bind` stricter than TS' own. I'm fine with merging this for now, but we should plan to revisit it before publishing types; in general my inclination is to *match* what TS does (sometimes even at the cost of strictness) for clean interoperability with the rest of the ecosystem.

Reply:
We could also add a catch-all type to allow more?

Reply:
I'm also open to loosening it up if people find enough issues in practice.

Reply:
Yeah, I go back and forth on `bind` and friends. In general I think we should probably also just get rid of *most* of the stuff in `@ember/runloop`, including this, but that’s a different discussion entirely. 😅